src/ui/utils/time.ts

```diff
@@ -9,3 +10,64 @@ export function formatTimestamp(ms: number) {
   const seconds = Math.round(ms / 1000.0);
   return `${Math.floor(seconds / 60)}:${padStart(String(seconds % 60), 2, "0")}`;
 }
+
+export function getDurationString(durationMs: number | null | undefined) {
+  if (typeof durationMs !== "number") {
+    return "";
+  }
+
+  return formatDuration(durationMs);
+}
+
+function shortenRelativeDate(date: string) {
+  return date
+    .replace("about", "")
+    .replace(" hours", "h")
+    .replace(" hour", "h")
+    .replace(" days", "d")
+    .replace(" day", "d")
+    .replace(" minutes", "m")
+    .replace(" minute", "m")
+    .replace(" seconds", "s")
+    .replace(" second", "s")
+    .replace("less than am", "1m");
+}
+
+export function getTruncatedRelativeDate(date: string, addSuffix = true) {
+  return getRelativeDate(date, true, addSuffix);
+}
+
+export function getRelativeDate(date: string, truncate: boolean = false, addSuffix = true) {
+  let content = formatDistanceToNow(new Date(date), { addSuffix });
+
+  if (truncate) {
+    content = shortenRelativeDate(content);
+  }
```

Comment:
nit: Nothing calls `getRelativeDate` directly, only `getTruncatedRelativeDate`, and nothing overrides the default param values.

```suggestion
export function getRelativeDate(date: string) {
  let content = shortenRelativeDate(
    formatDistanceToNow(new Date(date), { addSuffix: true })
  );
```